### PR TITLE
feat: Monero enhance handling decompression with std vector

### PR DESCRIPTION
**Pull request by thanhdanh**
# Summary 
- Optimize code of decode base64: Check the case of input length is zero + remove returning string
- Add Z_STREAM_ERROR in error handling of decompress

### CODE REVIEW

**Comment by thanhdanh** [comment]
@kewde I found another reason why we use `char *`. I see this warning in react-native bridge lib

Screenshot 2024-05-22 at 03 35 32

std::vector<char> is a C++ specific type and can cause conflict between C and C++ linkage conventions. There is an extern "C" which is used to specify that the linkage for the enclosed declarations is to be compatible with C, and C does not support C++ classes like std::vector.

File: src/serial_bridge_index.cpp [hunk 1]
```diff
@@ -128,19 +126,18 @@ const char* serial_bridge::decompress(const char *buffer, size_t length) {
                 throw std::runtime_error("Decompression error");
         }
 
-        decompressedData.insert(decompressedData.end(), outBuffer, outBuffer + sizeof(outBuffer) - strm.avail_out);
-    } while (ret == Z_OK);
+        decompressedData.insert(decompressedData.end(), outBuffer, outBuffer + BUFFER_SIZE - strm.avail_out);
+    } while (strm.avail_out == 0);
```
Comment (kewde) [inline src/serial_bridge_index.cpp:130]: Do we also want to add `Z_STREAM_ERROR` to the error switch statement?
https://refspecs.linuxbase.org/LSB_3.0.0/LSB-Core-generic/LSB-Core-generic/zlib-inflate-1.html

**Review by kewde — COMMENTED** [review]
Do we also want to add `Z_STREAM_ERROR` to the error switch statement?
https://refspecs.linuxbase.org/LSB_3.0.0/LSB-Core-generic/LSB-Core-generic/zlib-inflate-1.html

**Comment by thanhdanh** [comment]
> Do we also want to add `Z_STREAM_ERROR` to the error switch statement? https://refspecs.linuxbase.org/LSB_3.0.0/LSB-Core-generic/LSB-Core-generic/zlib-inflate-1.html

Sure, I missed it. The Z_STREAM_ERROR status code indicates that the stream state was inconsistent. Adding this case to the error handling will ensure that the function can gracefully handle such issues.

Added.

**Comment by thanhdanh** [comment]
tACK
- tested in react-native-fast-crypto

**Pull request merged**